### ISSUE TITLE
Unable to `phpactor-copy-class` "Trying to create descendant from absolute path.."

### ISSUE THREAD

**Issue by MarTango** [emacs-php/phpactor.el#12]
```
Could not copy file "/Users/martintang/test/test-php-project/src/MyClass.php" to "/Users/martintang/test/test-php-project/src/MyOtherClass.php"

0: Trying to create descendant from absolute path "/Users/martintang/test/test-php-project/src/MyClass.php" that does not lie within context path "/Users/martintang/test/test-php-project/"
#0 /Users/martintang/tools/phpactor/vendor/phpactor/source-code-filesystem/lib/Adapter/Git/GitFilesystem.php(70): Phpactor\Filesystem\Domain\FilePath->makeAbsoluteFromString(Object(Phpactor\Filesystem\Domain\FilePath))
#1 /Users/martintang/tools/phpactor/lib/Extension/ClassMover/Application/ClassCopy.php(94): Phpactor\Filesystem\Adapter\Git\GitFilesystem->createPath('/Users/martinta...')
#2 /Users/martintang/tools/phpactor/lib/Extension/ClassMover/Application/ClassCopy.php(83): Phpactor\Extension\ClassMover\Application\ClassCopy->doCopyFile(Object(Phpactor\Extension\ClassMover\Application\Logger\NullLogger), '/Users/martinta...', '/Users/martinta...')
#3 /Users/martintang/tools/phpactor/lib/Extension/ClassMover/Application/ClassCopy.php(51): Phpactor\Extension\ClassMover\Application\ClassCopy->copyFile(Object(Phpactor\Extension\ClassMover\Application\Logger\NullLogger), '/Users/martinta...', '/Users/martinta...')
#4 /Users/martintang/tools/phpactor/lib/Extension/ClassMover/Rpc/ClassCopyHandler.php(53): Phpactor\Extension\ClassMover\Application\ClassCopy->copy(Object(Phpactor\Extension\ClassMover\Application\Logger\NullLogger), '/Users/martinta...', '/Users/martinta...')
#5 /Users/martintang/tools/phpactor/lib/Extension/Rpc/RequestHandler/RequestHandler.php(39): Phpactor\Extension\ClassMover\Rpc\ClassCopyHandler->handle(Array)
#6 /Users/martintang/tools/phpactor/lib/Extension/Rpc/RequestHandler/ExceptionCatchingHandler.php(25): Phpactor\Extension\Rpc\RequestHandler\RequestHandler->handle(Object(Phpactor\Extension\Rpc\Request))
#7 /Users/martintang/tools/phpactor/lib/Extension/Rpc/RequestHandler/LoggingHandler.php(39): Phpactor\Extension\Rpc\RequestHandler\ExceptionCatchingHandler->handle(Object(Phpactor\Extension\Rpc\Request))
#8 /Users/martintang/tools/phpactor/lib/Extension/Rpc/Command/RpcCommand.php(74): Phpactor\Extension\Rpc\RequestHandler\LoggingHandler->handle(Object(Phpactor\Extension\Rpc\Request))
#9 /Users/martintang/tools/phpactor/lib/Extension/Rpc/Command/RpcCommand.php(62): Phpactor\Extension\Rpc\Command\RpcCommand->processRequest(Object(Phpactor\Extension\Rpc\Request))
#10 /Users/martintang/tools/phpactor/vendor/symfony/console/Command/Command.php(264): Phpactor\Extension\Rpc\Command\RpcCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#11 /Users/martintang/tools/phpactor/vendor/symfony/console/Application.php(874): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#12 /Users/martintang/tools/phpactor/vendor/symfony/console/Application.php(228): Symfony\Component\Console\Application->doRunCommand(Object(Phpactor\Extension\Rpc\Command\RpcCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#13 /Users/martintang/tools/phpactor/lib/Application.php(42): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#14 /Users/martintang/tools/phpactor/vendor/symfony/console/Application.php(130): Phpactor\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#15 /Users/martintang/tools/phpactor/bin/phpactor(43): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#16 {main}

1: Could not copy file "/Users/martintang/test/test-php-project/src/MyClass.php" to "/Users/martintang/test/test-php-project/src/MyOtherClass.php"
#0 /Users/martintang/tools/phpactor/lib/Extension/ClassMover/Application/ClassCopy.php(51): Phpactor\Extension\ClassMover\Application\ClassCopy->copyFile(Object(Phpactor\Extension\ClassMover\Application\Logger\NullLogger), '/Users/martinta...', '/Users/martinta...')
#1 /Users/martintang/tools/phpactor/lib/Extension/ClassMover/Rpc/ClassCopyHandler.php(53): Phpactor\Extension\ClassMover\Application\ClassCopy->copy(Object(Phpactor\Extension\ClassMover\Application\Logger\NullLogger), '/Users/martinta...', '/Users/martinta...')
#2 /Users/martintang/tools/phpactor/lib/Extension/Rpc/RequestHandler/RequestHandler.php(39): Phpactor\Extension\ClassMover\Rpc\ClassCopyHandler->handle(Array)
#3 /Users/martintang/tools/phpactor/lib/Extension/Rpc/RequestHandler/ExceptionCatchingHandler.php(25): Phpactor\Extension\Rpc\RequestHandler\RequestHandler->handle(Object(Phpactor\Extension\Rpc\Request))
#4 /Users/martintang/tools/phpactor/lib/Extension/Rpc/RequestHandler/LoggingHandler.php(39): Phpactor\Extension\Rpc\RequestHandler\ExceptionCatchingHandler->handle(Object(Phpactor\Extension\Rpc\Request))
#5 /Users/martintang/tools/phpactor/lib/Extension/Rpc/Command/RpcCommand.php(74): Phpactor\Extension\Rpc\RequestHandler\LoggingHandler->handle(Object(Phpactor\Extension\Rpc\Request))
#6 /Users/martintang/tools/phpactor/lib/Extension/Rpc/Command/RpcCommand.php(62): Phpactor\Extension\Rpc\Command\RpcCommand->processRequest(Object(Phpactor\Extension\Rpc\Request))
#7 /Users/martintang/tools/phpactor/vendor/symfony/console/Command/Command.php(264): Phpactor\Extension\Rpc\Command\RpcCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#8 /Users/martintang/tools/phpactor/vendor/symfony/console/Application.php(874): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#9 /Users/martintang/tools/phpactor/vendor/symfony/console/Application.php(228): Symfony\Component\Console\Application->doRunCommand(Object(Phpactor\Extension\Rpc\Command\RpcCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#10 /Users/martintang/tools/phpactor/lib/Application.php(42): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#11 /Users/martintang/tools/phpactor/vendor/symfony/console/Application.php(130): Phpactor\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#12 /Users/martintang/tools/phpactor/bin/phpactor(43): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#13 {main}
```

**Comment by zonuexe** [comment]
Hmm...  I tried to check the operation on both the absolute path and the relative path, but either one did not work.  I will investigate the implementation of Phpactor in a couple of days.

```
key::source_path value:/Users/megurine/repo/php/phpactor/lib/Container/InvalidConfig.php input:nil
key::dest_path value:nil input:lib/Container/AAa.php
(:action copy_class :parameters (:source_path /Users/megurine/repo/php/phpactor/lib/Container/InvalidConfig.php :dest_path lib/Container/AAa.php))
user-error: Could not copy file "/Users/megurine/repo/php/phpactor/lib/Container/InvalidConfig.php" to "/Users/megurine/work/contest-cms/src/ContestCms/Model/lib/Container/AAa.php"
```

**Comment by kermorgant** [comment]
Hi,

I stumbled upon the same issue but reported it to phpactor [in this issue](https://github.com/phpactor/phpactor/issues/479), and finally found a way to get rid of this error by [removing a trailing slash from the project root](https://github.com/phpactor/source-code-filesystem/pull/4).

The nice side effect of this is that it enables class autocomplete 
![image](https://user-images.githubusercontent.com/10561580/40565273-846e939c-6074-11e8-9362-24821ca186aa.png)

That said, I suspect a cleaner approach would be to remove the trailing slash from the working directory sent when doing the rpc call in emacs.

**Comment by kermorgant** [comment]
@ejmr 

decribe-fonts 
```
name (opened by): -ADBO-Source Code Pro-normal-normal-normal-*-22-*-*-*-m-0-iso10646-1
       full name: Source Code Pro:pixelsize=22:foundry=ADBO:weight=normal:slant=normal:width=normal:spacing=100:scalable=true
       file name: /usr/share/fonts/adobe-source-code-pro/SourceCodePro-Regular.otf
```

I believe it was the font recommended during the install of spacemacs

**Comment by zonuexe** [comment]
@kermorgant This problem has been fixed in #22.  Thanks a lot!

**Comment by zonuexe** [comment]
Maybe this issue is solved.  Thank you!
